frontend/src/components/forms/UserForm.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useEffect } from "react";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { Input } from "../ui/input";
+import { Button } from "../ui/button";
+import DatePicker from "../ui/DatePicker";
+import DietaryTypeSelector from "../templates/dietTemplates/DietaryTypeSelector";
+import { IUser } from "@/interfaces/IUser";
+
+const datePresets=[
+    {name:`חודש`, timeInDays:`30`},
+    {name:`חודשיים`, timeInDays:`60`},
+    {name:`שלושה חודשים`, timeInDays:`90`},
+    {name:`חצי שנה`, timeInDays:`180`},
+]
+
+const userSchema = z.object({
+  firstName: z.string().min(1, { message: "אנא הכנס שם פרטי" }),
+  lastName: z.string().min(1, { message: "אנא הכנס שם משפחה" }),
+  phone: z.string().min(7, { message: "אנא הכנס מספר טלפון תקין" }),
```

Comment:
why not do regex like the backend. for example one of the phone numbers is not complete in the current list of users. 

its like 052441333 its missin one number